### PR TITLE
ci(fp8): fix FP8 precision tests

### PR DESCRIPTION
**Pull request by CyCle1024**
## Summary
- Add replay-based k-grouped GEMM FP8 coverage against a FP32 scaled-grouped-mm reference.
- Tighten FP8 MoE train-engine loss references.
- Update the unit-test CI image to the current FP8-capable image.

### CODE REVIEW

File: tests/engine/test_moe_train_engine_float8.py [hunk 1]
```diff
@@ -178,10 +178,16 @@ def warmup_fn(x):
             lr_scheduler.step()
             losses.append(loss_log["reduced_llm_loss"])
 
-        losses_ref = torch.tensor([2.45, 2.45, 1.78, 1.31, 0.95, 0.67, 0.45, 0.31, 0.18, 0.12])
+        rank = dist.get_rank()
+        counter = 0
+        while os.path.exists(f"loss_fp8_{rank}_{counter}.txt"):
```
Comment (jayhenry) [inline tests/engine/test_moe_train_engine_float8.py]: remove debug file, or put it to /tmp/

**Review by jayhenry — APPROVED** [review]
